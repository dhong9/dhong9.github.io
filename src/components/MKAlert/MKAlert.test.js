import renderer from "react-test-renderer";
import MKAlert from "components/MKAlert";

// Mocks
jest.mock("components/MKBox", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock Box</div>)
    };
});
jest.mock("components/MKAlert/MKAlertRoot", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>MKAlertRoot</div>)
    };
});
jest.mock("components/MKAlert/MKAlertCloseIcon", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>MKAlertCloseIcon</div>)
    };
});
jest.mock("@mui/material/Fade", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Fade</div>)
    };
});

describe("MKAlert", () => {
    it("renders 'mount' status", () => {
        const component = renderer.create(<MKAlert />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders 'fadeOut' status", () => {
        const component = renderer.create(<MKAlert mount="fadeOut" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders default status", () => {
        const component = renderer.create(<MKAlert mount="default" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});