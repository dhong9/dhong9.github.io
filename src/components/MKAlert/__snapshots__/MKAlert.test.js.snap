// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKAlert renders 'fadeOut' status 1`] = `
<div>
  Fade
</div>
`;

exports[`MKAlert renders 'mount' status 1`] = `
<div>
  Fade
</div>
`;

exports[`MKAlert renders default status 1`] = `
<div>
  Fade
</div>
`;
